src/test/java/uk/ac/ebi/ena/readtools/webin/cli/rawreads/refs/CramReferenceInfoTest.java

```diff
@@ -7,13 +7,15 @@
 import java.util.Map;
 
 import org.junit.Assert;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import uk.ac.ebi.ena.readtools.webin.cli.rawreads.BamScannerTest;
 
 public class 
 CramReferenceInfoTest 
 {
+    @Ignore("Fails with an invalid sequence name exception. Sequence name validation rules have been updated in updated versions of htsjdk.")

```

Comment:
Why does the test fail? Is it because the CRAM is unaligned or because there is something wrong with the read names in the CRAM? Can we use another CRAM file to make the test work?

Reply:
the error that gets thrown is:

Sequence name '*' doesn't match regex: '[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*'

Newer version of htsjdk added this sequence name regex that was not present before. The cram file that is being tested here contains a (or more) sequence that has an invalid name. Good idea about using a different file for the test. You think can point me to small enough unaligned cram file?

Reply:
Is it possible to ignore this sequence name check in htsjdk? We don't index information about CRAMs being unaligned/aligned so we would need to loop through small CRAM files to find files that are unaligned. It would be very useful actually to have at least one one such file around that we can read with htsjdk as it might come handy in other tests as well later.

Reply:
Unfortunately it is not possible to skip/turn off this valdiation check. I tried looking in htsjdk's source code as well as searched the internet but found nothing. I also tried to see if it might be quicker to just edit the cram file and change the sequence name to something else but the tools I know also use htsjdk underneath and they also threw the same error when I attempted to use them. There might be another way to avoid this by using Java Reflection API to change the regex pattern that htsjdk uses but it might require a little RnD on my part.

Reply:
Managed to use Java Reflection API to set a dummy matcher for the test.